bemserver_core/model/timeseries.py

```diff
@@ -551,6 +529,41 @@ def register_class(cls):
         )
 
 
+def init_db_timeseries_triggers():
+    """Create triggers to protect some columns from update.
+
+    This function is meant to be used for tests or dev setups after create_all.
+    Production setups should rely on migration scripts.
+    """
+
+    # Set "update read-only trigger" on
+    #  campaign_id and campaign_scope_id columns for Timeseries table.
+    for ro_colname in [Timeseries.campaign_id.key, Timeseries.campaign_scope_id.key]:
+        db.session.execute(
+            generate_ddl_trigger_readonly(Timeseries.__table__, ro_colname)
+        )
+
+    # Set "update read-only trigger" on value_type column for timeseries property table.
+    db.session.execute(
+        generate_ddl_trigger_readonly(
+            TimeseriesProperty.__table__,
+            TimeseriesProperty.value_type.key,
+        )
+    )
+
+    # Set "update read-only trigger" on
+    #  timeseries_id and property_id columns for TimeseriesPropertyData table.
+    for ro_colname in [
+        TimeseriesPropertyData.timeseries_id.key,
+        TimeseriesPropertyData.property_id.key,
+    ]:
+        db.session.execute(
+            generate_ddl_trigger_readonly(TimeseriesPropertyData.__table__, ro_colname)
+        )
+
+    db.session.commit()
```

Comment:
This function could be 6 LoC by putting each call on a single line, no loop, no comment. No blocker. I may do it later.

Reply:
I factorized the logic in a new `make_columns_read_only` function to make those calls more readable.